File: Echo/FDRightDetailWithTextFieldCell.m
//
//  FDRightDetailWithTextFieldCell.m
//  Echo
//
//  Created by William Entriken on 2/6/14.
//
//

#import "FDRightDetailWithTextFieldCell.h"

@implementation FDRightDetailWithTextFieldCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        [self setup];
    }
    return self;
}

- (id)initWithCoder:(NSCoder *)aDecoder
{
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self setup];
    }
    return self;
}

- (void)awakeFromNib
{
    [super awakeFromNib];
    [self setup];
}

- (void)setup
{
    self.detailTextLabel.hidden = YES;
    [[self.contentView viewWithTag:3] removeFromSuperview];
    self.textField = [[UITextField alloc] initWithFrame:self.detailTextLabel.frame];
    self.textField.tag = 3;
    self.textField.textAlignment = NSTextAlignmentRight;
//    self.textField.translatesAutoresizingMaskIntoConstraints = NO;
    [self.contentView addSubview:self.textField];
/*
 [self addConstraint:[NSLayoutConstraint constraintWithItem:self.textField attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:self.textLabel attribute:NSLayoutAttributeTrailing multiplier:1 constant:8]];
    [self addConstraint:[NSLayoutConstraint constraintWithItem:self.textField attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.contentView attribute:NSLayoutAttributeTop multiplier:1 constant:8]];
    [self addConstraint:[NSLayoutConstraint constraintWithItem:self.textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self.contentView attribute:NSLayoutAttributeBottom multiplier:1 constant:-8]];
    [self addConstraint:[NSLayoutConstraint constraintWithItem:self.textField attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:self.detailTextLabel attribute:NSLayoutAttributeTrailing multiplier:1 constant:0]];
    [self removecon]
 */
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    // homeless approach
    self.textField.frame = CGRectMake(self.textLabel.frame.origin.x + self.textLabel.frame.size.width,
                                      self.detailTextLabel.frame.origin.y,
                                      self.detailTextLabel.frame.origin.x + self.detailTextLabel.frame.size.width
                                      - self.textLabel.frame.origin.x - self.textLabel.frame.size.width,
                                      self.detailTextLabel.frame.size.height);
}

@end

File: Echo/FDRightDetailWithTextFieldCell.swift
//
//  FDRightDetailWithTextFieldCell.swift
//  Echo
//
//  Created by William Entriken on 1/11/16.
//
//

//TODO: put this motherfucker straight into CocoaPods

import Foundation
import UIKit

class FDRightDetailWithTextFieldCell: UITableViewCell {
    var textField: UITextField!
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.setup()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.setup()
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.setup()
    }
    
    func setup() {
        self.detailTextLabel!.hidden = true
        self.contentView.viewWithTag(3)?.removeFromSuperview()
        self.textField = UITextField()
        self.textField.tag = 3
        self.textField.translatesAutoresizingMaskIntoConstraints = false
        self.contentView.addSubview(self.textField)
        self.addConstraint(NSLayoutConstraint(item: self.textField, attribute: .Leading, relatedBy: .Equal, toItem: self.contentView, attribute: .Leading, multiplier: 1, constant: 50))
        self.addConstraint(NSLayoutConstraint(item: self.textField, attribute: .Top, relatedBy: .Equal, toItem: self.contentView, attribute: .Top, multiplier: 1, constant: 8))
        self.addConstraint(NSLayoutConstraint(item: self.textField, attribute: .Bottom, relatedBy: .Equal, toItem: self.contentView, attribute: .Bottom, multiplier: 1, constant: -8))
        self.addConstraint(NSLayoutConstraint(item: self.textField, attribute: .Trailing, relatedBy: .Equal, toItem: self.contentView, attribute: .Trailing, multiplier: 1, constant: -16))
        self.textField.textAlignment = .Right
    }
    
    override func touchesEnded(touches: Set<UITouch>, withEvent event: UIEvent?) {
        self.textField.becomeFirstResponder()
    }
}